spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteTablePathsAction.java

```diff
@@ -1209,6 +1213,191 @@ public void testNestedDirectoryStructurePreservation() throws Exception {
     assertThat(targetPath2).startsWith(targetTableLocation());
   }
 
+  @Test
+  public void testFullRewriteUpdatesAllManifestLengthsInManifestList(
+      @TempDir Path rootTargetLocation) throws Exception {
+    String location = newTableLocation();
+    Table sourceTable = createTableWithSnapshots(location, 10);
+
+    addAnyPositionDelete(
+        sourceTable, removePrefix(sourceTable.location() + "/data/deeply/nested/deletes.parquet"));
+
+    Map<String, Long> manifestSizesBeforeRewrite =
+        sourceTable.currentSnapshot().allManifests(sourceTable.io()).stream()
+            .collect(Collectors.toMap(m -> fileName(m.path()), m -> m.length()));
```

Comment:
nit: not relate to your change, but I just realized for filename(m.path()), we can just replace with just `Paths.get(path).getFileName().toString()` and and `m.length()` can be replaced with Method reference